Hiding pdb_*_init would also hide pdb_search_init which is a public
interface used at least by pdbedit.

Index: source3/wscript_build
--- source3/wscript_build.orig
+++ source3/wscript_build
@@ -132,7 +132,9 @@ ldapsam_pdb_match = ['!priv2ld', '!smbldap_search_doma
                      '!get_attr_list']
 private_pdb_match.append('!pdb_nds_*')
 private_pdb_match.append('!pdb_ldapsam_init*')
-private_pdb_match.append('!pdb_*_init')
+private_pdb_match.append('!pdb_samba_dsdb_init')
+private_pdb_match.append('!pdb_smbpasswd_init')
+private_pdb_match.append('!pdb_tdbsam_init')
 private_pdb_match = private_pdb_match + ldapsam_pdb_match
 
 bld.SAMBA3_LIBRARY('samba-passdb',
